### PR TITLE
CoAP API style changes

### PR DESCRIPTION
**Pull request by otaviobp**
Fix issue #1720

Signed-off-by: Otavio Pontes <otavio.pontes@intel.com>

### CODE REVIEW

File: src/lib/comms/sol-coap.c [hunk 1]
```diff
@@ -1072,7 +1072,7 @@ sol_coap_packet_new_request(sol_coap_method_t method, sol_coap_msgtype_t type)
 }
 
 SOL_API int
-sol_coap_add_option(struct sol_coap_packet *pkt, uint16_t code, const void *value, uint16_t len)
+sol_coap_add_option(struct sol_coap_packet *pkt, uint16_t code, const void *value, size_t len)
```
Comment (vcgomes) [inline src/lib/comms/sol-coap.c]: The use of `uint16_t` documents that the option format only allows an option length that can be represented in 16 bits (the actual max would be 65804 IIRC, but I didn't see much value in supporting that).

If you really want to change this, also update `coap_option_encode()` and `coap_parse_option()` to take this into account.

File: src/lib/comms/include/sol-coap.h [hunk 2]
```diff
@@ -78,7 +78,7 @@ extern "C" {
  *
  * Refer to RFC 7252, section 12.2 for more information.
  */
-typedef enum {
+enum sol_coap_option_num_t {
```
Comment (vcgomes) [inline src/lib/comms/include/sol-coap.h]: Please remove the `_t` suffix as well, I don't think it's used for enums.

Comment (glima) [inline src/lib/comms/include/sol-coap.h]: ugh, -1 as well. if you were really following our patterns you'd remove the typedef (no _t)

File: src/lib/comms/sol-coap.c (hunk 1 above)
Comment (glima) [inline src/lib/comms/sol-coap.c]: I also prefer to document on .h's for those with 16bit limits then to change the type

File: src/lib/comms/include/sol-coap.h (hunk 2 above)
Comment (glima) [inline src/lib/comms/include/sol-coap.h]: And now I see that the enum part was done :)

File: bindings/nodejs/src/functions/oic-client-resource-ops.cc [hunk 3]
```diff
@@ -44,7 +44,7 @@ class OicClientRequest: public OicCallbackData {
     bool isOneShot;
 };
 
-static void requestAnswered(void *data, sol_coap_responsecode_t code,
+static void requestAnswered(void *data, sol_coap_response_code_t code,
```
Comment (cabelitos) [inline bindings/nodejs/src/functions/oic-client-resource-ops.cc]: I would also remove the `_t` suffix.

File: src/lib/comms/sol-coap.c (hunk 1 above)
Comment (barbieri) [inline src/lib/comms/sol-coap.c]: keep `uint16_t`

**Comment by otaviobp** [comment]
merged with fixes

**Pull request closed without merging**